src/main/java/com/shop/model/ProductObservation.java

```diff
@@ -35,19 +37,39 @@ public void setId(long id) {
         this.id = id;
     }
 
-    public User getUser() {
-        return user;
+    public UserConnection getUserConnection() {
+        return userConnection;
+    }
+
+    public void setUserConnection(UserConnection userConnection) {
+        this.userConnection = userConnection;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public Long getProductId() {
+        return productId;
+    }
+
+    public void setProductId(Long productId) {
+        this.productId = productId;
+    }
+
+    public Long getStartWatchTimestamp() {
+        return startWatchTimestamp;
     }
 
-    public void setUser(User user) {
-        this.user = user;
+    public void setStartWatchTimestamp(Long startWatchTimestamp) {
+        this.startWatchTimestamp = startWatchTimestamp;
     }
 
-    public Product getProduct() {
-        return product;
+    public Long getEndWatchTimestamp() {
+        return endWatchTimestamp;
     }
 
-    public void setProduct(Product product) {
-        this.product = product;
+    public void setEndWatchTimestamp(Long endWatchTimestamp) {
+        this.endWatchTimestamp = endWatchTimestamp;
     }
 }
```

Comment:
This class was supposed to store informations about user product observations like "I like this product, I want to sometimes look at the price and I will maybe buy it" but now it stores informations about session. What this class is supposed to do?

Reply:
discussed